week4/src/Header.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import styled from "styled-components";
+
+const Header = () => {
+  const [input, setInput] = useState("");
+  const navigate = useNavigate();
+
+  const onChange = (e) => {
+    setInput(e.target.value);
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    navigate(`/search/${input}`);
+  };
+
+  return (
+    <HeaderBlock>
+      <HeaderWrapper>Github Profile Finder </HeaderWrapper>
+      <SearchBar onSubmit={onSubmit}>
+        <SearchBlock
+          type="text"
+          value={input}
+          onChange={onChange}
+        ></SearchBlock>
+        <SearchButton type="submit">go</SearchButton>
+      </SearchBar>
+    </HeaderBlock>
+  );
+};
+
+export default Header;
+
+const HeaderBlock = styled.header`
+  margin: 0 auto;
+  border-radius: 10px;
+  background-color: #84b9bf;
+  color: white;
+  border: none;
+  /* height: 7rem; */
+  width: 65%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
```

Comment:
```suggestion
const HeaderBlock = styled.header`
  margin: 0 auto;
  border-radius: 10px;
  background-color: #84b9bf;
  color: white;
  border: none;
  /* height: 7rem; */
  width: 65%;
  display: flex;
  flex-direction: column;
  align-items: center;
  
  & > h1{
  text-align: center;
  } 
`;
```
으로 쓰면 밑에서 HeaderWrapper 안만들어도 데!